src/sentry/sentry_metrics/indexer/models.py

```diff
@@ -50,3 +58,146 @@ class Meta:
         constraints = [
             models.UniqueConstraint(fields=["string", "organization_id"], name="unique_org_string"),
         ]
+
+    @classmethod
+    def get_many_ids(cls, keys) -> Mapping[str, str]:
+        # keys => "organization_id:string"
+        final_results = {}
+        cache_results = indexer_cache.get_many(keys, field_type=str)
+        # {"key" => result}
+        db_look_up_keys = set()
+        for key in keys:
+            cache_key = indexer_cache.make_cache_key(key, field_type=str)
+            result = cache_results.get(cache_key)
+            if not result:
+                db_look_up_keys.add(key)
+
+            final_results[key] = result
+
+        cache_hits = len(final_results.keys())
+        cache_misses = len(db_look_up_keys)
+        # todo q: should this have an extra tag to know its from the write path?
+        metrics.incr(_INDEXER_CACHE_METRIC, tags={"cache_hit": "true"}, amount=cache_hits)
+        metrics.incr(_INDEXER_CACHE_METRIC, tags={"cache_hit": "false"}, amount=cache_misses)
+
+        if not db_look_up_keys:
+            return final_results
+
+        db_results = cls.get_db_ids_and_set_cache(db_look_up_keys)
+        final_results.update(db_results)
+
+        return final_results
+
+    @classmethod
+    def record_many_strings(cls, keys):
+        records = []
+        for key in keys:
+            organization_id, string = key.split(":")
+            records.append(cls(organization_id=int(organization_id), string=string))
+
+        with metrics.timer("sentry_metrics.indexer.pg_bulk_create"):
+            cls.objects.bulk_create(records, ignore_conflicts=True)
+
+        return cls.get_db_ids_and_set_cache(keys)
+
+    @classmethod
+    def get_db_ids_and_set_cache(cls, keys):
+        org_ids = []
+        strings = []
+        for key in keys:
+            organization_id, string = key.split(":")
+            org_ids.append(int(organization_id))
+            strings.append(string)
+
+        db_objects = cls.objects.filter(
+            organization_id__in=org_ids,
+            string__in=strings,
+        )
+        db_results = {}
+        reverse_db_results = {}
+        for db_result in db_objects:
+            key = f"{db_result.organization_id}:{db_result.string}"
+            if key in keys:
+                db_results[key] = db_result.id
+                reverse_key = f"{db_result.organization_id}:{db_result.id}"
+                reverse_db_results[reverse_key] = db_result.string
+
+        indexer_cache.set_many(db_results, field_type=str)
+        indexer_cache.set_many(reverse_db_results, field_type=int)
+        return db_results
+
+    @classmethod
+    def get_id(cls, org_id, string) -> int:
+        key = f"{org_id}:{string}"
+        result = indexer_cache.get(key, field_type=str)
+        if result:
+            metrics.incr(_INDEXER_CACHE_METRIC, tags={"cache_hit": "true"})
```

Comment:
Please add a tag to identify the method used to access the cache (`get_id` vs `get_many_ids` .....).